Add documentation for --build-profile warnings

cdmihai opened the pull request:
Added a page that details the incremental warnings and how users can address them.

dnfclas commented:
Hi __@cdmihai__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by .NET Foundation and real humans are currently evaluating your PR.</p>

TTYL, DNFBOT;

piotrpMSFT commented:
@blackdwarf can you review please?

blackdwarf commented on Documentation/addressing-incremental-compilation-warnings.md:
```diff
@@ -0,0 +1,17 @@
+
+The presence of such cases will turn off incremental compilation.
+
+Warning codes for project structures that are unsafe:
```
When you say "warning codes", this are the errors that the tool will write out?

cdmihai replied:
That's correct, they are the error codes the tool will print out. I'll update the doc to make it clear.

blackdwarf commented:
LGTM, with one minor nit.

cdmihai commented:
Addressed feedback

cdmihai commented:
Addressed feedback:

- tests are run in parallel
- separated passing the bin dir to test in another PR: #748
- rewrote the incremental tests to use the new testing framework

Currently blocked by #748. Can merge once that is in.
